### PR TITLE
Add notes to Trainer docs when devices flag is not defined

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add notes to Trainer docs when `devices` flag is not defined with accelerator

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

File: docs/source/common/trainer.rst [hunk 1]
```diff
@@ -241,6 +241,24 @@ Example::
 
     Trainer(accelerator=MyOwnAcc())
 
+.. note::
+    If the `devices` flag is not defined, it will assume `devices`` to be `"auto"` and fetch the `auto_device_count`
+    from the accelerator.
```
Comment (rohitgr7) [inline docs/source/common/trainer.rst:246]:
```suggestion
.. note::

    If the ``devices`` flag is not defined, it will assume ``devices``` to be ``"auto"`` and fetch the ``auto_device_count``
    from the accelerator.
```

File: docs/source/common/trainer.rst [hunk 2]
```diff
@@ -241,6 +241,24 @@ Example::
...
+
+.. code-block:: python
+
+    class GPUAccelerator(Accelerator):
```
Comment (rohitgr7) [inline docs/source/common/trainer.rst:250]:
```suggestion
    # this is part of built-in GPUAccelerator
    class GPUAccelerator(Accelerator):
```

File: docs/source/common/trainer.rst [hunk 3]
```diff
@@ -579,6 +597,24 @@ based on the accelerator type (``"cpu", "gpu", "tpu", "ipu", "auto"``).
     # Training with IPU Accelerator using 4 ipus
     trainer = Trainer(devices="auto", accelerator="ipu")
 
+.. note::
+    If the `devices` flag is not defined, it will assume `devices`` to be `"auto"` and fetch the `auto_device_count`
```
Comment (rohitgr7) [inline docs/source/common/trainer.rst:601]: I think this is already mentioned in L-584 as a tip .

**Review by rohitgr7 — APPROVED** [review]

File: docs/source/common/trainer.rst (hunk 3 above)
Comment (kaushikb11) [reply docs/source/common/trainer.rst:602]: I think it makes sense to mention it for both `accelerator` and `devices` section

File: docs/source/common/trainer.rst [hunk 4]
```diff
@@ -241,6 +241,25 @@ Example::
 
     Trainer(accelerator=MyOwnAcc())
 
+.. note::
+
+    If the ``devices`` flag is not defined, it will assume ``devices`` to be ``"auto"`` and fetch the ``auto_device_count``
+    from the accelerator.
+
+.. code-block:: python
```
Comment (rohitgr7) [inline docs/source/common/trainer.rst:249]: I think if you indent it, then it will appear under the note section.
like this: https://torchmetrics.readthedocs.io/en/latest/pages/lightning.html

File: docs/source/common/trainer.rst (hunk 3 above)
Comment (kaushikb11) [reply docs/source/common/trainer.rst:602]: Merge both in what ways?

File: docs/source/common/trainer.rst (hunk 4 above)
Comment (kaushikb11) [reply docs/source/common/trainer.rst:249]: It was intentional, don't want under `notes` section

Comment (rohitgr7) [reply docs/source/common/trainer.rst:249]: why is that? isn't the example reflecting what's mentioned in the note?

File: docs/source/common/trainer.rst (hunk 3 above)
Comment (kaushikb11) [reply docs/source/common/trainer.rst:603]: I think it makes sense to keep it separate.

Comment (rohitgr7) [reply docs/source/common/trainer.rst:603]: also back-ticks needs to be fixed :)

Comment (kaushikb11) [reply docs/source/common/trainer.rst]: Fixed!

**Review by rohitgr7 — APPROVED** [review]

File: docs/source/common/trainer.rst [hunk 5]
```diff
@@ -579,6 +599,26 @@ based on the accelerator type (``"cpu", "gpu", "tpu", "ipu", "auto"``).
     # Training with IPU Accelerator using 4 ipus
     trainer = Trainer(devices="auto", accelerator="ipu")
 
+.. note::
+
+    If the ``devices`` flag is not defined, it will assume ``devices`` to be ``"auto"`` and fetch the ``auto_device_count``
```
Comment (Borda) [inline docs/source/common/trainer.rst:604]: is it the same on the same page?

Comment (kaushikb11) [reply docs/source/common/trainer.rst:604]: Yup

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]
